tel2puml/__main__.py

```diff
@@ -252,17 +276,82 @@ def generate_component_options(
     return otel_pv_options, pv_puml_options
 
 
+def handle_exception(
+    e: Exception,
+    debug: bool,
+    user_error: bool = False,
+    custom_message: str = "",
+    exit_code: int = 1,
+) -> None:
+    """Handle exceptions with custom messaging and exit codes.
+
+    :param e: The exception instance to handle.
+    :type e: :class:`Exception`
+    :param debug: Flag to indicate if debug information should be printed.
+    :type debug: `bool`
+    :param user_error: Flag to indicate if the error is a user error,
+    defaults to False.
+    :type user_error: `bool`, optional
+    :param custom_message: Custom error message for user, defaults to "".
+    :type custom_message: `str`, optional
+    :param exit_code: Exit code for the program, defaults to 1.
+    :type exit_code: `int`, optional
+    """
+    if debug:
+        print(f"DEBUG: {traceback.format_exc()}")
+    else:
+        print("\nERROR: Use the -d flag for more detailed information.")
+        if user_error:
+            print(f"User error: {custom_message} {e}")
+        else:
+            print(f"An unexpected error occurred. {e}. Please"
+                  " contact smartDCSIT support for assistance.")
+
+    exit(exit_code)
+
+
 if __name__ == "__main__":
     warnings.filterwarnings("ignore")
 
     args: argparse.Namespace = parser.parse_args()
     args_dict = vars(args)
-    otel_pv_options, pv_puml_options = generate_component_options(
-        args.command, args_dict
-    )
-    otel_to_puml(
-        otel_pv_options,
-        pv_puml_options,
-        args_dict["output_file_directory"],
-        args.command,
-    )
+    debug = args_dict.get("debug", False)
```

Comment:
Could you actually take al this code and put it into a `main(args_dict: dict[str, Any], errors_lookup: dict[Type[Exception], TypedDict[]])` function so that we can test the error handling